### ISSUE TITLE
Autocomplete input should be type="search"

### ISSUE THREAD

**Issue by coliff** [trimble-oss/modus-web-components#2044]
### Prerequisites

- [X] I have [searched](https://github.com/trimble-oss/modus-web-components/issues?q=is%3Aissue) for duplicate or closed issues
- [X] I have read the [contributing guidelines](https://github.com/trimble-oss/modus-web-components/blob/main/CONTRIBUTING.md)

### Describe the issue

Entering text in Autocomplete and pressing Escape should clear the input as any default type="search" text would. If the input was type=search and not type=text then this would work as expected.

This requires: #524

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Microsoft Edge

### What is the issue regarding ?

@trimble-oss/modus-web-components

### What version of npm package are you using ?

v.0.12.0

### Priority

Medium

### What product/project are you using Modus Components for ?

Modus

### What is your team/division name ?

Modus

### Are you willing to contribute ?

Maybe

### Are you using Modus Web Components in production ?

https://modus-web-components.trimble.com/
